sdk/python/kfp/_client.py

```diff
@@ -219,7 +220,8 @@ def _load_config(self, host, client_id, namespace, other_client_id, other_client
       token = get_gcp_access_token()
       self._is_refresh_token = False
     elif credentials:
-      token = credentials.get_token()
+      config.api_key['authorization'] = 'placeholder'
+      config.api_key_prefix['authorization'] = 'Bearer'
```

Comment:
Not a blocker for this PR, from https://github.com/kubeflow/pipelines/pull/5945 context, we should not add bearer authorization either.

Reply:
In fact, I think we should write a base class for bearer authorization.
Then e.g. others can write a base class for basic authentication

Reply:
@Bobgy unfortunately I don't think we can do this.

If you check the links I've included in https://github.com/kubeflow/pipelines/issues/5138#issuecomment-797003710 explaining the `Configuration`'s codebase, `refresh_api_key_hook` is only triggered when `authorization` is in `api_key` ([source](https://github.com/kubernetes-client/python/blob/f17076f0e12bd894877825906864b42f25756f0c/kubernetes/client/configuration.py#L332)).

Reply:
That's a great catch! I didn't know the limitation. Let's leave that use case in the backlog and get this PR merged first.